photon-ml/src/main/scala/com/linkedin/photon/ml/function/L1RegularizationTerm.scala

```diff
@@ -1,22 +0,0 @@
-/*
- * Copyright 2016 LinkedIn Corp. All rights reserved.
- * Licensed under the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License. You may obtain a
- * copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
-package com.linkedin.photon.ml.function
-
-/**
- * A trait to represent whether an L1 regularization is applied to a [[DiffFunction]] or [[TwiceDiffFunction]].
- */
-trait L1RegularizationTerm {
-  def getL1RegularizationParam: Double
-}
```

Comment:
@joshvfleming 
L1Regularization trait is gone now; treated as a special case when selecting an optimizer.